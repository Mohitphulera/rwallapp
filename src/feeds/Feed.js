import React, { Component } from 'react'
import Post from '../components/Post'
import { request } from '../services/request'

export default class Feed extends Component {
    constructor(props) {
        super(props)
        this.state = {
            posts: null,
            notify: false
        }
    }

    componentDidMount() {
        this.retrievePosts()
    }

    render() {
        const { posts, notify } = this.state
        let items = null
        if (posts && posts.length > 0) {
            items = posts.map(post => <Post key={post.id} post={post} />)
        } else if (notify) {
            items = <div className="notification is-info" style={{opacity: "0.8"}}>There's no posts to display!</div>
        }
        return (
            <section className="section home">
                <div className="container">{items}</div>
            </section>
        )
    }

    retrievePosts() {
        request
            .get('posts/')
            .then(posts => {
                if (posts) {
                    this.setState({
                        posts,
                        notify: false
                    })
                }
            })
            .catch(() => {
                console.error('Request for posts failed.')
                this.setState({
                    notify: true
                })
            })
    }
}
