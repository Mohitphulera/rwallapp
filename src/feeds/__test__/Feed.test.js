import React from 'react'
import ReactDom from 'react-dom'
import Feed from './../Feed'
import {isTSAnyKeyword} from  '@babel/types'
import renderer from "react-test-renderer"

it("renders without crashing",()=> {
    const div = document.createElement("div");
    ReactDom.render(<Feed />,div)

})

it("matches snapshot & renders correctly",() => {
    const tree=renderer.create(<Feed/>).toJSON();
    expect(tree).toMatchSnapshot();
})