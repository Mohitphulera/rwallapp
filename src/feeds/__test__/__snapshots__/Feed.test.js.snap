// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot & renders correctly 1`] = `
<section
  className="section home"
>
  <div
    className="container"
  />
</section>
`;
